sdl/cpu.go

```diff
@@ -0,0 +1,48 @@
+package sdl
+
+import (
+	"sort"
+
+	"github.com/pkg/errors"
+	"github.com/xtgo/set"
+	"gopkg.in/yaml.v3"
+)
+
+type v2CPUAttributes v2ResourceAttributes
+
+type v2ResourceCPU struct {
+	Units      cpuQuantity     `yaml:"units"`
+	Attributes v2CPUAttributes `yaml:"attributes,omitempty"`
+}
+
+func (sdl *v2CPUAttributes) UnmarshalYAML(node *yaml.Node) error {
+	attr := make(v2CPUAttributes)
+
+	for i := 0; i < len(node.Content); i += 2 {
```

Comment:
`i+1 < len(node.Content)`

Reply:
:point_up:

Reply:
missed that comment. shouldn't be necessary, or i missed thing 😄

Reply:
defensive programming.  it's not clear why the yaml would always be a map.